deepchem/models/torch_models/layers.py

```diff
@@ -4723,3 +4724,71 @@ def message(self, x_i: torch.Tensor, x_j: torch.Tensor,
                         dim=0)
 
         return x_j
+
+
+class MXMNetBesselBasisLayer(torch.nn.Module):
+    """This layer implements a basis layer for the MXMNet model using Bessel functions.
+    The basis layer is used to model radial symmetry in molecular systems.
+
+    The output of the layer is given by:
+    output = envelope(dist / cutoff) * (freq * dist / cutoff).sin()
+
+    Examples
+    --------
+    >>> radial_layer = MXMNetBesselBasisLayer(num_radial=2, cutoff=2.0, envelope_exponent=2)
+    >>> distances = torch.tensor([0.5, 1.0, 2.0, 3.0])
+    >>> output = radial_layer(distances)
+    >>> output.shape
+    torch.Size([4, 2])
+    """
+
+    def __init__(self,
+                 num_radial: int,
+                 cutoff: float = 5.0,
+                 envelope_exponent: int = 5):
+        """Initialize the MXMNet Bessel Basis Layer.
+
+        Parameters
+        ----------
+        num_radial: int
+            The number of radial basis functions to use.
+        cutoff: float, optional (default 5.0)
+            The radial cutoff distance used to scale the distances.
+        envelope_exponent: int, optional (default 5)
+            The exponent of the envelope function.
+        """
+
+        super(MXMNetBesselBasisLayer, self).__init__()
+        self.cutoff = cutoff
+        self.envelope: _MXMNetEnvelope = _MXMNetEnvelope(envelope_exponent)
```

Comment:
It should be fine here now, but try to use general type hints in cases like these like `nn.Module`